pkgs/by-name/li/libstaden-read/package.nix

```
@@ -0,0 +1,52 @@
+{ lib
+, stdenv
+, autoreconfHook
+, fetchFromGitHub
+, gcc
+, bzip2
+, xz
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+  # Same name as the Debian library
+  pname = "libstaden-read";
+  version = "1.15.0";
+  src = fetchFromGitHub {
+    owner = "jkbonfield";
+    repo = "io_lib";
+    rev = "io_lib-" + builtins.replaceStrings ["."] ["-"] version;
+    fetchSubmodules = true;
+    sha256 = "sha256-2Dlx+MXmqar81/Xmf0oE+6lWX461EDYijiZsZf/VD28=";
+  };
+
+  patches = [
+    # Needed so that the lib can be detected
+    ./libstaden-install-config-header.patch
+  ];
+
+  buildInputs = [ gcc zlib xz bzip2 ];
+  nativeBuildInputs = [ autoreconfHook ];
+
+  # autoreconfHook does not descend into htscodecs folder
+  preAutoreconf = ''
+    pushd ./htscodecs
+    autoreconf --install --force --verbose
+    pushd
+  '';
+
+  meta = {
+    description = "C library for reading/writing various DNA sequence formats";
+    longDescription = ''
+    Libstaden-read is a library of file reading and writing code to provide a
+    general purpose SAM/BAM/CRAM, trace file (and Experiment File) reading
+    interface. Programmatically {S,B,CR}AM can be manipulated using the
+    scram_*() API functions while DNA Chromatogram ("trace") files can be read
+    using the read_reading() function.'';
+    homepage = "https://staden.sourceforge.net";
+    downloadPage = "https://github.com/jkbonfield/io_lib/releases";
```

TIL about [`downloadPage`](https://nixos.org/manual/nixpkgs/unstable/#var-meta-downloadPage). You should also define `changelog` which is more popular (assuming you use `finalAttrs`).

```suggestion
    downloadPage = "https://github.com/jkbonfield/io_lib/releases";
    changelog = "https://github.com/jkbonfield/io_lib/blob/${finalAttrs.src.rev}/CHANGES";
```